toolchain/check/testdata/builtins/int_add.carbon

```diff
@@ -0,0 +1,231 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// --- int_add.carbon
+
+fn Add(a: i32, b: i32) -> i32 = "int.add";
+
+var arr: [i32; Add(1, 2)];
+
+fn RuntimeCall(a: i32, b: i32) -> i32 {
+  return Add(a, b);
+}
+
+// --- fail_bad_decl.carbon
+
+package FailBadDecl api;
+
+fn TooFew(a: i32) -> i32 = "int.add";
+fn TooMany(a: i32, b: i32, c: i32) -> i32 = "int.add";
+fn BadReturnType(a: i32, b: i32) -> bool = "int.add";
+fn JustRight(a: i32, b: i32) -> i32 = "int.add";
+
+// CHECK:STDERR: fail_bad_decl.carbon:[[@LINE+3]]:20: ERROR: Array bound is not a constant.
+// CHECK:STDERR: var too_few: [i32; TooFew(1)];
+// CHECK:STDERR:                    ^~~~~~~
+var too_few: [i32; TooFew(1)];
+// CHECK:STDERR: fail_bad_decl.carbon:[[@LINE+3]]:21: ERROR: Array bound is not a constant.
+// CHECK:STDERR: var too_many: [i32; TooMany(1, 2, 3)];
+// CHECK:STDERR:                     ^~~~~~~~
+var too_many: [i32; TooMany(1, 2, 3)];
+// CHECK:STDERR: fail_bad_decl.carbon:[[@LINE+3]]:28: ERROR: Array bound is not a constant.
+// CHECK:STDERR: var bad_return_type: [i32; BadReturnType(1, 2)];
+// CHECK:STDERR:                            ^~~~~~~~~~~~~~
+var bad_return_type: [i32; BadReturnType(1, 2)];
+
+// CHECK:STDERR: fail_bad_decl.carbon:[[@LINE+6]]:21: ERROR: 3 argument(s) passed to function expecting 2 argument(s).
+// CHECK:STDERR: var bad_call: [i32; JustRight(1, 2, 3)];
+// CHECK:STDERR:                     ^~~~~~~~~~
+// CHECK:STDERR: fail_bad_decl.carbon:[[@LINE-18]]:1: Calling function declared here.
+// CHECK:STDERR: fn JustRight(a: i32, b: i32) -> i32 = "int.add";
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+var bad_call: [i32; JustRight(1, 2, 3)];
+
```

Comment:
```suggestion
var bad_call: [i32; JustRight(1, 2, 3)];

fn WrapTooFew(a: i32) -> i32 {
  return TooFew(a);
}

fn WrapTooMany(a: i32) -> i32 {
  return TooMany(a, a, a);
}

fn WrapBadReturnType(a: i32) -> i32 {
  return BadReturnType(a, a);
}

```

You're testing cases with constant parameters, should it also be tested with non-constant parameters?

Reply:
Sure, added with a TODO to diagnose these before we get to lower.